File: codeforces/110A/go/110a.go
package main

import "fmt"


func is_lucky(n int) bool {
    if n == 0 {
        return false
    }

    for n != 0 {
        mod := n % 10
        n /= 10
        if mod != 4 && mod != 7 {
            return false
        }
    }

    return true
}

func main() {
    var num string
    fmt.Scan(&num)

    n := 0
    for i := 0; i < len(num); i++ {
        if num[i] == '4' || num[i] == '7' {
            n++
        }
    }

    if (is_lucky(n)) {
        fmt.Println("YES")
    } else {
        fmt.Println("NO")
    }
}

File: codeforces/110A/cpp/110a.cpp
#include <iostream>
using namespace std;

typedef unsigned long long u64;

bool is_lucky(u64 num)
{
    if (num == 0)
        return false;

    while (num) {
        u64 mod = num % 10;
        num /= 10;
        if (mod != 4 && mod != 7)
            return false;
    }

    return true;
}

bool is_nearly_lucky(u64 num)
{
    if (num == 0)
        return false;

    int n = 0;
    while (num) {
        u64 mod = num % 10;
        num /= 10;
        if (mod == 4 || mod == 7)
            n++;
    }

    return is_lucky(n);
}

int main()
{
    u64 num;
    cin >> num;

    if (is_nearly_lucky(num)) {
        cout << "YES" << endl;
    }
    else {
        cout << "NO" << endl;
    }

    return 0;
}
